common/buildcraft/lib/net/MessageUpdateTile.java

```diff
@@ -46,6 +49,7 @@ public void fromBytes(ByteBuf buf) {
     public void toBytes(ByteBuf buf) {
         PacketBuffer buffer = new PacketBuffer(buf);
         buffer.writeBlockPos(pos);
+        buffer.writeString(className);
```

Comment:
i had that bug as well in BCA, instead of a string i gave every TE an a number and used that to sync, a short should be enough